packages/app-layout/src/AppLayoutContent.tsx

```diff
@@ -0,0 +1,125 @@
+import { Fragment, PropsWithChildren } from 'react';
+import { Prose } from '@ag.ds-next/react/prose';
+import { PageContent } from '@ag.ds-next/react/content';
+import { Callout } from '@ag.ds-next/react/callout';
+import {
+	boxPalette,
+	CoreProvider,
+	mapSpacing,
+	Spacing,
+	tokens,
+} from '@ag.ds-next/react/core';
+import { Text } from '@ag.ds-next/react/text';
+import { TextLink } from '@ag.ds-next/react/text-link';
+
+export type ExpectedClaims = {
+	name?: string;
+	given_name?: string;
+	family_name?: string;
+
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	[others: string]: any;
+};
+
+const isEmptyString = (t: string | undefined) => (t?.trim() ?? '') === '';
+
+const Divider = (props: { spacing?: Spacing }) => (
+	<hr
+		css={{
+			boxSizing: 'content-box',
+			height: 0,
+			overflow: 'visible',
+			border: 'none',
+			borderTopWidth: tokens.borderWidth.sm,
+			borderTopStyle: 'solid',
+			borderColor: boxPalette.borderMuted,
+			width: '100%',
+			marginTop: mapSpacing(props?.spacing ?? 2),
+			marginBottom: mapSpacing(props?.spacing ?? 2),
+		}}
+	/>
+);
+
+// This exists as its own package in ag.common.
+// Likely cleaner to re-use it here but makes consuming it more error prone (peer deps).
+const HelpCallout = () => {
+	return (
+		<CoreProvider>
+			<Callout title="Need more help?">
+				<Text>
+					Email{' '}
+					<TextLink href="mailto:exportservice@aff.gov.au">
+						exportservice@aff.gov.au
+					</TextLink>
+				</Text>
+				<Text>
+					Call <TextLink href="tel:1800571125">1800&nbsp;571&nbsp;125</TextLink>
+					, Monday to Friday, 9 am to 5 pm AEST
+				</Text>
+			</Callout>
+		</CoreProvider>
+	);
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const MissingName = (props: PropsWithChildren) => (
+	<PageContent>
+		<Prose>
+			<h1>Single names in the Export Service</h1>
+			<p>
+				If you are legally known by a single name, you can use it to access an
+				Export Service account. Other services we offer may not accept a single
+				name and you will need to use a given and family name.
+			</p>
+			<p>Contact support if you need more help.</p>
+
+			<Divider />
+			<HelpCallout />
+		</Prose>
+	</PageContent>
+);
+
+const MissingGivenName = MissingName;
+const MissingFamilyName = MissingName;
+
+export const AppErrorComponents = {
+	MissingGivenName,
+	MissingFamilyName,
+};
+
+export type ErrorComponents = typeof AppErrorComponents;
+
+// Business logic goes here. We prevent app access by default if claims look wrong.
+// This behaviour can be overridden by providing appropriate error components.
+export const AppContent = ({
+	claims,
+	errorComponents: ErrorComponents,
+	children,
+}: PropsWithChildren<{
+	claims?: ExpectedClaims;
+	errorComponents: ErrorComponents;
+}>) => {
+	// If no claims provided, just act naturally. This preserves compatibility.
+	if (claims === undefined) {
+		return <Fragment>{children}</Fragment>;
```

Comment:
Can remove the `<Fragment>`s in these conditions?

Reply:
last time i removed fragments we had some strange build issue on the other end when I went to consume components from ag.common so I am just defensively placing them everywhere now. I still dont remember why it was a problem.


https://github.com/agriculturegovau/ag-common/commit/c65ab89adc9e2ace4ace3cd43bce7d8ccbf6be62

Reply:
Weird